### PR TITLE
codon: 0.15.5 -> 0.16.1

### PR DESCRIPTION
**Pull request by paveloom**
###### Description of changes

https://github.com/exaloop/codon/releases/tag/v0.16.1

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by risicle** [comment]
On macos 10.15 I got:

```
error: hash mismatch in fixed-output derivation '/nix/store/54vpprgbsig2g0qmmc0h4hnbn5qa5ljk-codon-deps-0.16.0.tar.gz.drv':
         specified: sha256-a1zGSpbMjfQBrcgW/aiIdKX8+uI3p/S9pgZjHe2HtWs=
            got:    sha256-KfemYV42xBAhsPbwTkzdc3GxCVHiWRbyUZORPWxx4vg=
```

**Comment by paveloom** [comment]
@risicle Thanks! Turns out the dependencies are different on Darwin systems.

I tried it in a virtual machine, and now I'm getting

```
./result/bin/codon --version
dyld: Library not loaded: /nix/store/vq59agppci8ll3n7bfgyzkdx08gyhp0m-codon-0.16.0/lib/libcodonc.dylib
  Referenced from: /Users/paveloom/Playground/nixpkgs/./result/bin/codon
  Reason: image not found
Abort trap: 6
```

Seems like there is a misbehaved symbolic link created somewhere.

**Comment by paveloom** [comment]
Aight, that's fixed now, too.

File: pkgs/development/compilers/codon/default.nix [hunk 1]
```diff
@@ -117,21 +120,22 @@ stdenv.mkDerivation {
...
+  '';
+
   passthru.updateScript = nix-update-script { };
 
   meta = {
     description = "A high-performance, zero-overhead, extensible Python compiler using LLVM";
     homepage = "https://docs.exaloop.io/codon";
     maintainers = [ lib.maintainers.paveloom ];
-    license = lib.licenses.bsl11;
+    # license = lib.licenses.bsl11;
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/codon/default.nix:138]: What's up with this?

Comment (paveloom) [reply pkgs/development/compilers/codon/default.nix]: Oh, right, forgot to remove after testing (the license is unfree so it's easier to comment the line).

File: pkgs/development/compilers/codon/default.nix [hunk 2]
```diff
@@ -88,7 +86,12 @@ let
...
     '';
 
-    outputHash = "sha256-a1zGSpbMjfQBrcgW/aiIdKX8+uI3p/S9pgZjHe2HtWs=";
+    outputHash =
+      if stdenv.hostPlatform.isDarwin then
+        "sha256-KfemYV42xBAhsPbwTkzdc3GxCVHiWRbyUZORPWxx4vg="
+      else
+        "sha256-a1zGSpbMjfQBrcgW/aiIdKX8+uI3p/S9pgZjHe2HtWs=";
+
     outputHashAlgo = "sha256";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/codon/default.nix:95]:
```suggestion
```
no longer required with sri

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**